pyiceberg/table/update/snapshot.py

```diff
@@ -445,6 +462,14 @@ def files_affected(self) -> bool:
         """Indicate if any manifest-entries can be dropped."""
         return len(self._deleted_entries()) > 0
 
+    def _validate(self, current_metadata: TableMetadata, snapshot: Optional[Snapshot]) -> None:
+        if snapshot is None:
+            raise CommitFailedException("Snapshot cannot be None.")
```

Comment:
I think this is not correct. When you create an empty table, there is no current-snapshot.